src/coreclr/jit/importer.cpp

```diff
@@ -5464,6 +5464,32 @@ GenTree* Compiler::impOptimizeCastClassOrIsInst(GenTree* op1, CORINFO_RESOLVED_T
     return nullptr;
 }
 
+bool Compiler::impMatchIsInstBooleanConversion(const BYTE* codeAddr, const BYTE* codeEndp, int* consumed)
+{
+    OPCODE nextOpcode = impGetNonPrefixOpcode(codeAddr, codeEndp);
+    switch (nextOpcode)
+    {
+        case CEE_BRFALSE:
+        case CEE_BRFALSE_S:
+            *consumed = 0;
+            return true;
+        case CEE_BRTRUE:
+        case CEE_BRTRUE_S:
+            *consumed = 0;
+            return true;
+        case CEE_LDNULL:
+            nextOpcode = impGetNonPrefixOpcode(codeAddr + 1, codeEndp);
+            if (nextOpcode == CEE_CGT_UN)
```

Comment:
can be CEQ as well? not sure if it's popular

Reply:
I had it initially, but it requires reversing the condition and I didn't manage to get Roslyn to ever produce it, so I decided to get rid of it to simplify the logic. I just ran SPMI replay and only saw a handful of test-only hits, so I think I just won't bother. Looks like `impBoxPatternMatch` doesn't handle it either.